content/plugins/pof-importer/pof-importer.php

```diff
@@ -135,309 +187,411 @@ public function import() {
+    }
 
-                    // has a parent
-                    if ( isset( $item['parent'] ) ) {
-                        $parent_guid = $item['parent'];
-                        $parent_id = $this->queried[$parent_guid][$item['lang']]->ID;
-                    } else {
-                        $parent_id = 0;
-                    }
+    /**
+     * Begin the actual import process
+     *
+     * @param array $tree Api data.
+     */
+    private function update_pages( &$tree ) {
+
+        // Create wp cli progress bar
+        $progress = $this->wp_cli_progress( 'Importing page content', count( $tree ) );
+        $start    = microtime( true );
+
+        // Update/Create invidual pages
+        foreach ( $tree as &$item ) {
+
+            // Collect connected posts
+            $translations = [];
+            foreach ( $item['languages'] as $lang ) {
+                if ( $lang['page'] ) {
+                    $translations[ $lang['lang'] ] = $lang['page']->ID;
+                    break;
+                }
+            }
+            $new_translations = false;
+
+            // Update translations
+            foreach ( $item['languages'] as &$lang ) {
+                if ( $lang['update_page'] && $lang['data'] ) {
+
+                    // Post object params
+                    $args = [
+                        'post_title'    => sanitize_title( $lang['title'] ),
+                        'menu_order'    => $item['order'],
+                        'post_status'   => 'publish',
+                        'post_type'     => 'page',
+                        'post_title'    => $item['title'],
+                        'post_content'  => $lang['data']['content'],
+                        'page_template' => 'models/page-' . $lang['data']['type'] . '.php',
+                        'post_modified' => $lang['lastModified'],
+                    ];
 
-                    // Default values for page creation/updation
-                    $args['post_status']   = 'publish';
-                    $args['post_type']     = 'page';
-                    $args['post_title']    = $item['title'];
-                    $args['post_parent']   = $parent_id;
-                    $args['post_content']  = isset($item['content']) ? $item['content'] : '';
-                    $args['page_template'] = 'models/page-' . $item['type'] . '.php';
-
-                    
-                    // Create the post
-                    $post_id = wp_insert_post( $args );
-
-                    if ( ! is_wp_error( $post_id ) && $post_id !== 0 && isset( $old_slug ) ) {
-                        $new_slug = wp_make_link_relative( get_permalink( $post_id ));
-                        if ( $new_slug !== $old_slug ) {
-                            $redirect_cache[ $old_slug ] = $new_slug;
+                    // If item has parent link to it
+                    if ( $item['parent'] ) {
+                        $parent_id = $this->queried[ $item['parent'] ][ $lang['lang'] ]->ID;
```

Comment:
Is there a possibility this could be empty? Maybe there should be null coalescing operator here?

Reply:
Added empty check.

Reply:
This is good but my comment might have been a bit vague. I meant the line 483. If the parent item id is not present in the queried array, it will cause an empty warning. I would change the line as follows:
```
$parent_id = $this->queried[ $item['parent'] ][ $lang['lang'] ]->ID ?? false;
```
This will silence all warnings for the left side.

I take that this will probably never cause a warning. No need to improve this further if you think this is sufficient.